src/MagnumExternal/TinyGltf/tiny_gltf.h

```diff
@@ -3686,46 +6141,91 @@ static void SerializeNumberProperty(const std::string &key, T number,
   // obj.insert(
   //    json_object_pair(key, json(static_cast<double>(number))));
   // obj[key] = static_cast<double>(number);
-  obj[key] = number;
+  JsonAddMember(obj, key.c_str(), json(number));
 }
 
 template <typename T>
 static void SerializeNumberArrayProperty(const std::string &key,
                                          const std::vector<T> &value,
                                          json &obj) {
-  json o;
-  json vals;
+  if (value.empty()) return;
 
-  for (unsigned int i = 0; i < value.size(); ++i) {
-    vals.push_back(static_cast<T>(value[i]));
-  }
-  if (!vals.is_null()) {
-    obj[key] = vals;
+  json ary;
+  JsonReserveArray(ary, value.size());
+  for (const auto &s : value) {
+    JsonPushBack(ary, json(s));
   }
+  JsonAddMember(obj, key.c_str(), std::move(ary));
 }
 
 static void SerializeStringProperty(const std::string &key,
                                     const std::string &value, json &obj) {
-  obj[key] = value;
+  JsonAddMember(obj, key.c_str(), JsonFromString(value.c_str()));
 }
 
 static void SerializeStringArrayProperty(const std::string &key,
                                          const std::vector<std::string> &value,
                                          json &obj) {
-  json o;
-  json vals;
-
-  for (unsigned int i = 0; i < value.size(); ++i) {
-    vals.push_back(value[i]);
+  json ary;
+  JsonReserveArray(ary, value.size());
+  for (auto &s : value) {
+    JsonPushBack(ary, JsonFromString(s.c_str()));
   }
-
-  obj[key] = vals;
+  JsonAddMember(obj, key.c_str(), std::move(ary));
 }
 
 static bool ValueToJson(const Value &value, json *ret) {
```

Comment:
Too bad tinygltf doesn't have a define for compiling out all the serialization stuff we don't need :/ I assume that is also what needed the json.hpp update?